lib/addHooks.js

```diff
@@ -91,26 +34,8 @@ Stack: ${error.stack}
     runner.hooks.transactions[transaction.name] = transaction;
   });
 
-  // Loading hooks from string, sandbox mode must be enabled
-  if (!(runner && runner.configuration && runner.configuration.options
-        && runner.configuration.options.hookfiles)) {
-    if (runner.configuration.hooksData) {
-      if (runner.configuration.options.sandbox === true) {
-        return loadSandboxHooksFromStrings(callback);
-      }
-      // Not sandboxed code can't be loaded from the string
-      const msg = `
-  Not sandboxed hooks loading from strings is not implemented,
-  Sandbox mode must be enabled when loading hooks from strings.
-`;
-      return callback(new Error(msg));
-    }
-    // No data found, doing nothing
-    return callback();
-  }
-
   // Loading hookfiles from fs
-  const hookfiles = [].concat(runner.configuration.options.hookfiles);
+  const hookfiles = [].concat(runner.configuration.options.hookfiles || []);
```

Comment:
I don't think there is such case in real world usage, but many existing tests didn't pass without it, because they don't properly init the configuration. So it seemed easier to do this fallback in the codebase than to debug all the (unit)tests and set hookfiles to empty array everywhere `configuration` is being used as an input 😞